pyfifotax/report_data.py

```diff
@@ -0,0 +1,616 @@
+                    )
+                )
+
+            elif isinstance(event, MoneyDepositEvent):
+                if event.amount > 0:
+                    new_forex = FIFOForex(
+                        currency=event.currency,
+                        quantity=event.amount,
+                        buy_date=event.buy_date,
+                        source=f"Deposit of Money",
+                    )
+                    self.held_forex[event.currency].push(new_forex)
+
+                if event.fees is not None:
+                    self.held_forex[event.fees.currency].pop(
+                        event.fees.amount,
+                        to_decimal(1),
+                        event.date,
+                    )
+
+                if event.fees is not None:
+                    self.misc["Fees"].append(event.fees)
+
+            elif isinstance(event, MoneyWithdrawalEvent):
+                if event.amount > 0.0:
+                    tmp = self.held_forex[event.currency].pop(
+                        event.amount,
+                        to_decimal(1),
+                        event.date,
+                    )
+                    self.withdrawn_forex[event.currency].extend(tmp)
+
+                if event.fees is not None:
+                    # first pop fees, later not tracked in withdrawn amount (only in fees)
+                    self.held_forex[event.currency].pop(
+                        event.fees.amount, to_decimal(1), event.date
+                    )
+                    self.misc["Fees"].append(event.fees)
+
+            elif isinstance(event, CurrencyConversionEvent):
+                sold_forex = self.held_forex[event.source_currency].pop(
+                    event.source_amount, to_decimal(1), event.date
+                )
+                self.sold_forex[event.source_currency].extend(sold_forex)
+
+                new_forex = FIFOForex(
+                    currency=event.target_currency,
+                    quantity=event.target_amount,
+                    buy_date=event.date,
+                    source=f"Currency Conversion {event.source_currency} to {event.target_currency}",
+                )
+                self.held_forex[event.target_currency].push(new_forex)
+
+                if event.fees is not None:
+                    self.held_forex[event.fees.currency].pop(
+                        event.fees.amount,
+                        to_decimal(1),
+                        event.date,
+                    )
+
+                # fees here considered as part of Fees in general (not as cost of transactions)
+                # as FOREX likely is not taxed as Capital Gain, thus it should be fine to
+                # simplify things here and consider them as potential Werbungskosten
+                if event.fees is not None:
+                    self.misc["Fees"].append(event.fees)
+
+            elif isinstance(event, StockSplitEvent):
+                if self.apply_stock_splits:
```

Comment:
What do you mean with wrong stock split data? stock_splits.xlsx should all contain all NVDA splits at least (if this is not the case, then this is indeed a bug).
I still would like to keep this flag because some people might have a history of manually maintained data-points and in the worst-case would have scaled things on their own.